File: neural_network.py
#!/usr/bin/env python3

import matplotlib.pyplot as plt
import numpy as np
import pandas as pd


sport_labels = {
    'Cycling': 0,
    'Swiss Wrestling': 1,
    'Alpine Skiing': 2,
    'Sumo Wrestling': 3,
    'Long-Distance Running': 4,
}
label_sports = {v: k for k, v in sport_labels.items()}
training_data = pd.read_csv('train.csv')

training_data['Label'] = training_data['Sport'].map(sport_labels)
print(training_data)

for label in training_data['Label'].unique():
    entries = training_data[training_data['Label'] == label]
    x = entries['Weight']
    y = entries['Height']
    plt.scatter(x, y, label=label_sports[label])
plt.legend()

plt.savefig('train.png')

File: octave/neural_network.m
% train.csv contains data about male athletes. Columns:
% 1) height in meters
% 2) weight in kilograms
% 3) sport as label
%   0: Cycling
%   1: Swiss Wrestling
%   2: Alpine Skiing
%   3: Sumo Wrestling
%   4: Long-Distance Running
data = csvread('train.csv');
X = data(:,[1,2]);
Y = data(:,3);

% normalize input data
X(:,1) = (X(:,1) - mean(X(:,1))) / std(X(:,1));
X(:,2) = (X(:,2) - mean(X(:,2))) / std(X(:,2));

% dimension information
m = length(X);
n = length(unique(Y));

% one-hot encode labels
y = zeros(m, n)
for i = 1:(max(Y)+1)
    y(:,i) = (Y == (i-1));
end

X
y
